lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,707 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: elb_lb_application
+short_description: Manage an Application load balancer
+description:
+    - Manage an AWS Application Elastic Load Balancer. See U(https://aws.amazon.com/blogs/aws/new-aws-application-load-balancer/) for details.
+version_added: "2.3"
+author: "Rob White (@wimnat)"
+options:
+  access_logs_s3_bucket:
+    description:
+      - The name of the S3 bucket for the access logs. This attribute is required if access logs in Amazon S3 are enabled. The bucket must exist in the same region as the load balancer and have a bucket policy that grants Elastic Load Balancing permission to write to the bucket.
+    required: false
+  access_logs_s3_prefix:
+    description:
+      - The prefix for the location in the S3 bucket. If you don't specify a prefix, the access logs are stored in the root of the bucket.
+    required: false
+  deletion_protection:
+    description:
+      - Indicates whether deletion protection for the ELB is enabled.
+    required: false
+    default: no
+    choices: [ 'yes', 'no' ]
+  idle_timeout:
+    description:
+      - The number of seconds to wait before an idle connection is closed.
+    required: false
+    default: 60
+  listeners:
+    description:
+      - A list of dicts containing listeners to attach to the ELB. See examples for a detail of the dict required.
+    required: false
+  name:
+    description:
+      - The name of the load balancer. This name must be unique within your AWS account, can have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, and must not begin or end with a hyphen.
+    required: true
+  purge_tags:
+    description:
+      - If yes, existing tags will be purged from the resource to match exactly what is defined by tags parameter. If the tag parameter is not set then tags will not be modified.
+    required: false
+    default: yes
+    choices: [ 'yes', 'no' ]
+  subnets:
+    description:
+      - A list of the IDs of the subnets to attach to the load balancer. You can specify only one subnet per Availability Zone. You must specify subnets from at least two Availability Zones. Required if state=present.
+    required: false
+  security_groups:
+    description:
+      - A list of the names or IDs of the security groups to assign to the load balancer. Required if state=present.
+    required: false
+    default: []
+  scheme:
```

Comment:
It may be more obvious to name this parameter something like `public` as a boolean, but I'd be worried about AWS adding some third scheme meaning we have to obsolete the boolean and use their new value. No change required here, but would like your thoughts.